src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitUsingTests.cs

```diff
@@ -2315,5 +2354,43 @@ void M()
                 Diagnostic(ErrorCode.ERR_NoConvToIAsyncDisp, "var y = new object()").WithArguments("object").WithLocation(7, 22)
                 );
         }
+
+        [Fact]
+        [WorkItem(30956, "https://github.com/dotnet/roslyn/issues/30956")]
+        public void GetAwaiterBoxingConversion()
+        {
+            var source =
+@"using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+struct StructAwaitable { }
+
+class Disposable
+{
+    public StructAwaitable DisposeAsync() => new StructAwaitable();
+}
+
+static class Extensions
+{
+    public static TaskAwaiter GetAwaiter(this object x)
+    {
+        if (x == null) throw new ArgumentNullException(nameof(x));
```

Comment:
>if (x == null) throw new ArgumentNullException(nameof(x)); [](start = 8, length = 58)

Consider printing something so that we can verify that the method is called.